File: truevfs-ext/truevfs-ext-pacemanager/src/main/java/net/java/truevfs/ext/pacemanager/PaceManagerDecorator.java
/*
 * Copyright (C) 2005-2012 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.pacemanager;

import javax.annotation.concurrent.Immutable;
import net.java.truevfs.kernel.spec.FsManager;
import net.java.truevfs.kernel.spec.spi.FsManagerDecorator;

/**
 * @deprecated This class is reserved for exclusive use by the
 *             {@link FsManagerLocator#SINGLETON}!
 * @author Christian Schlichtherle
 */
@Deprecated
@Immutable
public final class PaceManagerDecorator extends FsManagerDecorator {

    @Override
    public FsManager apply(FsManager manager) {
        return PaceMediator.SINGLETON.instrument(manager);
    }

    /** Returns 100. */
    @Override
    public int getPriority() {
        return 100;
    }
}

File: truevfs-ext/truevfs-ext-pacemanager/src/main/scala/net/java/truevfs/ext/pace/PaceManagerDecorator.scala
/*
 * Copyright (C) 2005-2012 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.pace

import net.java.truevfs.kernel.spec._
import net.java.truevfs.kernel.spec.spi._

/**
  * @author Christian Schlichtherle
  */
@deprecated("This class is reserved for exclusive use by the [[net.java.truevfs.kernel.spec.sl.FsManagerLocator.SINGLETON]]!", "1")
final class PaceManagerDecorator extends FsManagerDecorator {
  def apply(manager: FsManager): FsManager = PaceMediator.instrument(manager)

  /**Returns 100. */
  override def getPriority = 100
}
